### DISCUSSION TITLE
tracing for sqlmodel ?

### DISCUSSION THREAD

**Discussion by k1eran** [traceloop/openllmetry#1645] [Q&A] [answered]
Hi @nirga,
 
I have recently experimenting with `traceloop/openllmetry` and so far it looks great.

My Python app also uses [SQLModel](https://github.com/tiangolo/sqlmodel) for interacting with my SQL database.

Do you have any future plans for trace re. SQLModel, 
or any recommendations for a strategy for it that would work well with my existing traceloop usage ?

Cheers
Kieran.

**Comment by nirga** [comment]
Hey @k1eran! Looks like SQLModel is using SQLAlchemy behind the scenes, which is something we instrument. Were you able to see SQL-related spans?

**Reply by k1eran** [reply]
Hi @nirga -
Yes I can see connect operations spans related to sqlalchemy but without detail just the `connect` level.

**Comment by nirga** [comment]
Thanks @k1eran, I think this has to do with the behavior of the [SQLAlchemy instrumentation](https://opentelemetry-python-contrib.readthedocs.io/en/latest/instrumentation/sqlalchemy/sqlalchemy.html) provided by OpenTelemetry. We'll look into providing a better support for this, but if you can open an issue and provide a small snippet that can be extremely helpful 🙏

**Reply by k1eran** [reply, marked as answer]
@nirga When I enable my underlying db adapter (`PsycopgInstrumentor().instrument()`) used below SQLAlchemy I realise I now get pretty good tracing (with the SQL commands)

Thanks for pointing me in right direction.
Cheers. Kieran.

PS. I think I edited my earlier reply at same time you replied; I THINK this whole thread still makes sense however to others.
